refactor/restApiAlwaysReturnJSON (1 part)

DotsenkoS opened the pull request.

DotsenkoS committed:
Создан class ResponseDto<T> - обертка для ответов REST контроллевов.
Внесены изменения в AdminRestController - на все запросы возвращает ResponseDto<T>

DotsenkoS committed:
Merge branch 'dev' into refactor/restApiAlwaysReturnJSON

Dalan2692 committed:
refactor changeMail method

Dalan2692 committed:
create new method for changeMail

Dalan2692 committed:
create abstract method for changeMail in CustomerService

Dalan2692 committed:
refactor all endpoints of controller

Dalan2692 committed:
refactor all service methods

Dalan2692 committed:
add annotation @JsonIgnore for list of SharedNews

Dalan2692 committed:
change return value for deleteById

Dalan2692 committed:
create exceptionHandler to catch selfmade exceptions

Dalan2692 committed:
create exception for customerService

Dalan2692 committed:
create exception for news

Dalan2692 committed:
create CustomerDto fot Customer Entity

Dalan2692 committed:
create NewsDto fot News Entity

Dalan2692 committed:
refactor ModelMapper , create final instance for all endpoints

DotsenkoS committed:
Merge branch 'dev' into refactor/restApiAlwaysReturnJSON

Dalan2692 committed:
refactor  endpoints : change return type with DTO

Dalan2692 committed:
before changing return types of abstract methods

Dalan2692 committed:
create , refacor manager's endpoints

DotsenkoS committed:
Внесены изменения в AuthenticationManager.
Внесены изменения в CustomerRestController - основая часть контроллеров  на все запросы возвращает ResponseDto<T>.Оставшиеся- в процессе

DotsenkoS committed:
Merge branch 'dev' into refactor/restApiAlwaysReturnJSON

DotsenkoS committed:
Merge remote-tracking branch 'origin/refactor/rest_api_always_return_json_part_2' into refactor/restApiAlwaysReturnJSON

# Conflicts:
#	src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java
#	src/main/java/com/jm/online_store/controller/rest/manager/ManagerRestController.java

DotsenkoS committed:
Внесены изменения в AuthenticationManager.
Внесены изменения в CustomerRestController - основая часть контроллеров  на все запросы возвращает ResponseDto<T>.Оставшиеся- в процессе

DotsenkoS committed:
Внесены изменения в
 AdminRestController
 CustomerRestController
 CustomerTrackableProductRestController
 ModeratorRestController
 AuthenticationRestControllerJWT
 На все запросы контроллеры возвращают ResponseDto<T>
 Пустые строки игнорируются.
 Если ответ это пустой массив, то он не игнорируется и отдается как "data": []

tediik committed:
Update SharedStockService.java

tediik committed:
Update SharedStock.java

tediik committed:
Update StockServiceException.java

tediik committed:
Update SentStockServiceException.java

tediik committed:
Update OrderServiceException.java

tediik commented on src/main/java/com/jm/online_store/exception/newsService/NewsExceptionConstants.java:
```diff
@@ -0,0 +1,12 @@
+
+public class NewsExceptionConstants {
+
+    public final static String NO_NEWS_WITH_SUCH_ID = "There are no news with such id : %s ";
+    public final static String NO_NEWS_YET = "There are no news yet";
+    public final static String NO_PUBLISHED_NEWS = "There are no published news";
+    public final static String NO_UNPUBLISHED_NEWS = "There are no unpublished news";
+    public final static String NO_ARCHIVED_NEWS = "There are no archived news";
```
так, давай все все константы для exeption вынесем все же в 1 класс единый и положим в пакет exception

tediik commented on src/main/java/com/jm/online_store/controller/rest/manager/ManagerStockRestController.java:
```diff
@@ -32,33 +38,27 @@
 @Api(description = "Rest controller for manage of stocks from manager page")
 public class ManagerStockRestController {
     private final StockService stockService;
+    private final ModelMapper modelMapper = new ModelMapper();
```
так можно определить его просто как бин, и во ВСЕХ классах поключать с помощью Спринга
private final ModelMapper modelMapper;
Он сам через конструктор заинжектит

tediik commented on src/main/java/com/jm/online_store/controller/rest/manager/ManagerSharedStockRestController.java:
```diff
@@ -26,24 +33,28 @@
+            SharedStock sharedStockFromService = sharedStockService.addSharedStock(sharedStock);
+            returnValue = modelMapper.map(sharedStockFromService, SharedStockDto.class);
         } catch (Exception e) {
             return ResponseEntity.badRequest().build();
```
ну по идее тогда нужно и этот момент обработать в новом стиле

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -45,6 +53,8 @@
     private final CustomerService customerService;
     private final UserService userService;
     private final RecentlyViewedProductsService recentlyViewedProductsService;
+    @Autowired
+    private ModelMapper modelMapper;
```
почему не через конструктор как у нас общепринято в проекте?

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -53,90 +63,75 @@
-        if (customerService.changePassword(customer.getId(), oldPassword, newPassword))
-            log.debug("Изменения для пользователя с идентификатором: {} был успешно добавлен", customer.getId());
-        return ResponseEntity.ok().build();
+        customerService.changePassword(customer.getId(), oldPassword, newPassword);
+        log.debug("Changes for user with ID: {} added successfully", customer.getId());
+        return new ResponseEntity<>(new ResponseDto<>(true, "Изменения для пользователя с идентификатором: " + customer.getId() + " были успешно добавлены. ", ResponseOperation.NO_ERROR.getMessage()), HttpStatus.OK);
```
т.е. метод всегда отдает 200ОК, не важно, что пароль неверный, уже есть такой юзер и т.д.??
Или в сервисе вылетает exeption, который обработает ControllerAdvice и выкинет наружу уже нужный ответ?

tediik commented on src/main/java/com/jm/online_store/controller/rest/admin/AdminRestController.java:
```diff
@@ -44,139 +52,148 @@
 
     private final CommonSettingsService commonSettingsService;
 
+    @Autowired
+    private ModelMapper modelMapper;
```
через конструктор предпочтительнее

tediik committed:
Update ResponseOperation.java

tediik committed:
Update ExceptionsHandler.java

tediik commented on src/main/java/com/jm/online_store/service/impl/CustomerServiceImpl.java:
```diff
@@ -114,9 +117,12 @@ public void addCustomer(Customer customer) {
-        Customer customer = customerRepository.findById(id).orElseThrow(UserNotFoundException::new);
-        customer.setDayOfWeekForStockSend(null);
-        updateCustomer(customer);
+        Optional<Customer> optCustomer = customerRepository.findById(id);
+        if (optCustomer.isEmpty()) {
+            throw new CustomerServiceException(CustomerExceptionConstants.CUSTOMER_NOT_FOUND);
+        }
+        optCustomer.get().setDayOfWeekForStockSend(null);
+        updateCustomer(optCustomer.get());
```
все же, нельзя ли было сохранить подход через orElseThrow?

tediik commented on src/main/java/com/jm/online_store/service/impl/NewsServiceImpl.java:
```diff
@@ -57,21 +59,27 @@
     @Override
     public News findById(long id) {
-        return newsRepository.findById(id).orElseThrow(() -> new NewsNotFoundException("There are no news with such id"));
+        if (newsRepository.findById(id).isEmpty()) {
+             throw new NewsServiceException(String.format(NewsExceptionConstants.NO_NEWS_WITH_SUCH_ID, id));
+        }
+        return newsRepository.findById(id).get();
```
все же, нельзя ли было сохранить подход через orElseThrow?

tediik commented on src/main/java/com/jm/online_store/service/impl/NewsServiceImpl.java:
```diff
@@ -99,8 +107,13 @@ public News update(News news) {
      */
     @Override
-    public void deleteById(Long id) {
+    public boolean deleteById(Long id) {
+        Optional<News> optionalNews = newsRepository.findById(id);
+        if (optionalNews.isEmpty()) {
+            throw new NewsServiceException(String.format(NewsExceptionConstants.NO_NEWS_WITH_SUCH_ID, id));
+        }
```
через orElseThrow точно никак?

tediik reviewed (changes requested):
Обсудим на созвоне еще

DotsenkoS committed:
Внесены изменения в
 AdminRestController
 CustomerRestController
 CustomerTrackableProductRestController
 ModeratorRestController
 AuthenticationRestControllerJWT
 На все запросы контроллеры возвращают ResponseDto<T>
 Пустые строки игнорируются.
 Если ответ это пустой массив, то он не игнорируется и отдается как "data": []

DotsenkoS committed:
- ExceptionConstants и ExceptionEnums вынесены в отдельный класс.
- Исправлено по замечаниям.

DotsenkoS committed:
Merge remote-tracking branch 'origin/refactor/restApiAlwaysReturnJSON' into refactor/restApiAlwaysReturnJSON

tediik committed:
Update ManagerStockRestController.java

tediik committed:
Update ManagerNewsRestController.java

tediik committed:
Update ManagerRestController.java
